doc/integrator/authentication.rst

```diff
@@ -210,3 +213,108 @@ LDAP would look like this::
         if data is not None:
             return data[0]
         return default_user_validator(request, username, password)
+
+Custom user validation - LDAP
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Full example using pyramid_ldap
+
+.. code:: python
+
+    # -*- coding: utf-8 -*-
+
+    from pyramid.config import Configurator
+    from pyramid.security import authenticated_userid
+    from c2cgeoportal import locale_negotiator, add_interface, \
+        INTERFACE_TYPE_SENCHA_TOUCH, get_user_from_request
+    from c2cgeoportal.lib.authentication import create_authentication
+    from yourproject.resources import Root
+
+    import logging
+
+    import ldap
+    from json import dumps, loads
+
+    log = logging.getLogger(__name__)
+
+    def custom_user_validator(request, username, password):
+        from c2cgeoportal import default_user_validator
+        from pyramid_ldap import get_ldap_connector
+        connector = get_ldap_connector(request)
+        data = connector.authenticate(username, password)
+
+        if data is not None:
+            log.debug('user %s found in ldap' % username)
+            log.debug(pp.pformat(data[1]))
+            user = {'username': data[1]['uid'][0], 'role': data[1]['roles'][0]}
+            return dumps(user)
+
+        log.debug("user %s not found in ldap, searching locally" % username)
+        return default_user_validator(request, username, password)
+
+    def custom_get_user_from_request(request):
+
+        class O(object):
+            pass
+
+        from c2cgeoportal.models import DBSession, Role
+
+        user = get_user_from_request(request)
+        if user is None:
+            log.debug("user is not authenticated or is a ldap user")
+            identity = authenticated_userid(request)
+            if identity is not None:
+                identity = loads(identity)
+                user = O()
+                user.username = identity['username']
+                user.functionalities = []
+                user.is_password_changed = True
+                user.role = DBSession.query(Role).filter_by(name=identity['role']).one()
+                user.id = -1
+
+        return user
+
+    def main(global_config, **settings):
+        """ This function returns a Pyramid WSGI application.
+        """
+        config = Configurator(
+            root_factory=Root, settings=settings,
+            locale_negotiator=locale_negotiator,
+            authentication_policy=create_authentication(settings)
+        )
+
+        config.include("c2cgeoportal")
+        config.include('pyramid_ldap')
+
+        config.ldap_setup(
+            'ldap://ldap.server.host',
+            bind='CN=ldap user,CN=Users,DC=example,DC=com',
+            passwd='ld@pu5er'
```

Comment:
thats the fake password from pyramid_ldap http://docs.pylonsproject.org/projects/pyramid-ldap/en/latest/